### ISSUE TITLE
False recognition of 9

### ISSUE THREAD

**Issue by 1stDeadeye** [jomjol/AI-on-the-edge-device#441]
Hello,

first, thank you for this fantastic project!

I have the problem that the 9 is recognized as 0, although the image quality is good. I think the ROIs also sit very well.

Does anyone have a tip on what it could be?

Example: Digit number 6 is a "9", but recognized as "0":
![Screenshot 2021-12-04 204305](https://user-images.githubusercontent.com/6626117/144722808-5e928a54-9860-4a11-a24e-832e5d06b771.png)

ROIs configuration für digit 6:
![Screenshot 2021-12-04 203708](https://user-images.githubusercontent.com/6626117/144722863-ddc03467-320d-4c03-aeaf-57809da4efd1.png)

Original image:
![Download](https://user-images.githubusercontent.com/6626117/144722897-263e94b6-bed7-4b1c-b409-5640d20c0eb6.png)

**Comment by 1stDeadeye** [comment]
I have now switched to AlignmentAlgo = highAccurity and the recognition (seems) to work properly now. I continue to monitor this. If anyone has any tips, please let me know.

**Comment by jomjol** [comment]
If it is still not okay, than maybe I need to add your digits to the training. In this case please let me know.

**Comment by 1stDeadeye** [comment]
It is not perfect, but I think that is normal. In 80-90% of all cases, the digits are recognized correctly.
But I can collect all the digits and send them to you. I think the more numbers trained, the better for everyone.

When collecting the numbers, I take best the individual images of the recognition?

**Comment by jomjol** [comment]
Yes and please send only 2-3 image per digit (0, ... 9). Sometimes people send the full log files with 10000 of images. I cannot process that :-).

**Comment by 1stDeadeye** [comment]
I have now collected a few numbers. Hope this fits so far.
Let me know if you need anything else :-)

[numbers.zip](https://github.com/jomjol/AI-on-the-edge-device/files/7687837/numbers.zip)

**Comment by jomjol** [comment]
Hi, the format is "jtif", my script can only handle jpg or bmp. Can you please convert then, thanks!

**Comment by 1stDeadeye** [comment]
No problem :)

[numbers-jpg.zip](https://github.com/jomjol/AI-on-the-edge-device/files/7688312/numbers-jpg.zip)

**Comment by thost96** [comment]
Any news here? Was there an update? I'm facing the same issue on my gas meter.

**Comment by jomjol** [comment]
I'm working on it, but I have really problems to get the neural network training to converge with a good result.
I need to adjust the training / strucutre, but therefore I need some time. Maybe the christmas holidays.

**Comment by janecker** [comment]
Hi,

seems like I have a similar problem with detecting "9" resulting in 0.

![image](https://user-images.githubusercontent.com/1313022/146638963-cf1d505f-d896-431f-a2c3-9ea5b640233c.png)

![image](https://user-images.githubusercontent.com/1313022/146638989-0bff9212-1171-44ae-8879-9e23b70a1e59.png)


Is it fine to highjack this thread or should I open a new ticket for it?

**Comment by jomjol** [comment]
@janecker No, that is fine. Just send me some example images (2-3 images of each digit) and I will incorporate them in the next training session

**Comment by janecker** [comment]
@jomjol thanks, great. Any specific way of capturing the images, is it fine to save the digits from the "Recognition" tab? Any specific naming scheme you would prefer?

**Comment by cyberius0** [comment]
Hello,
same here. Number "9" is being recognized as "0".

<img width="357" alt="20211221 115257-jomjol - AI on the edge - watermeter — Mozilla Firefox" src="https://user-images.githubusercontent.com/3000648/146918464-452352b0-a45e-41ef-90f7-c4ea7b25e1bd.png">

**Comment by jomjol** [comment]
@cyberius0 @janecker @1stDeadeye @thost96 

Please test the follwing new neural network file (need to unzip) - it contains the latest set of images:
 
[dig-s0-q.zip](https://github.com/jomjol/AI-on-the-edge-device/files/7757879/dig-s0-q.zip)

**Comment by janecker** [comment]
@jomjol: Thank you, on the first look it seem to work now! I will check for some time and will give feedback if something comes up. :)

**Comment by janecker** [comment]
Thanks for the update. The 9 looks good now, but it seems to have a problem detecting the 6 now:

![image](https://user-images.githubusercontent.com/1313022/147142058-b851a285-293b-4933-beda-305614349afc.png)
![6_1](https://user-images.githubusercontent.com/1313022/147141939-79ec1924-e6d7-4863-b786-56f65e6c528c.jpg)

it is now detected as 0. Does it have trouble because the number started turning already towards the 7?

**Comment by jomjol** [comment]
Yes, I think so. With this setting, this is almost impossilbe to avoid, because there are also some "0", that have a smaller diameter and this is hard to distinguish from a shifted "6". It will be a smaller problem in one of the next versions, where I also will detect the intermediate state, which means, that in your case, the image will be more like a "6.2".
But this is still in development and also needs extended training sets for the digits.

**Comment by janecker** [comment]
Got it. Is there any workaround I can use to make it work? Otherwise I have to wait untill it reaches the 7 (which should not take too long..)

**Comment by jomjol** [comment]
Currently no way arround beside making your own training data.

**Comment by janecker** [comment]
I just checked on it again I got a false recognition again:

![image](https://user-images.githubusercontent.com/1313022/147363235-16518af6-46e0-419f-b65a-6ee62db68cc7.png)

As expected the leading 6 is still having problems, but also the 2 is detected as a 0 now. Here is the jpeg of the 2, just in case you want to give it another try:

![2_1](https://user-images.githubusercontent.com/1313022/147363320-3417765d-e6ac-4856-8edb-60ef5147f1af.jpg)

**Comment by janecker** [comment]
Another day another update. Again I have new false recognition:

![image](https://user-images.githubusercontent.com/1313022/147385235-362dbc08-aa07-4f33-b21b-f0cdadaf7ec3.png)

Here are the digits again:
![5_1](https://user-images.githubusercontent.com/1313022/147385260-114b48f8-a929-469d-a4c0-5bbdad685df3.jpg)
![9_1](https://user-images.githubusercontent.com/1313022/147385262-694c0dc1-ac6f-4f5c-9893-679080db9d49.jpg)

**Comment by jomjol** [comment]
@janecker : I have made a detailed description how to setup the ROIs and additional how to collect the images for further training: https://github.com/jomjol/AI-on-the-edge-device/wiki/ROI-Configuration

On demand please provide me training images and I will incorporate them.

**Comment by janecker** [comment]
@jomjol: Sorry for the late response and thank you very much for this new documentation. I was looking for exactly that when I started setting my system up as well when I started troubleshooting it. So I think this will be a big help. For now my problem is solved as my counter passed the 7000 now and the detection seems to work fine now.

P.S.: I mostly posted the updates in case this might help to improve the system, I was fine with waiting, to see if everything starts working fine after passing that stupid flip of the 6 :).
